micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java

```diff
@@ -315,14 +323,17 @@ else if (executorService instanceof ForkJoinPool) {
             monitor(registry, (ForkJoinPool) executorService);
         }
         else if (allowIllegalReflectiveAccess) {
+            // For Executors.newSingleThreadScheduledExecutor()
             if (className.equals("java.util.concurrent.Executors$DelegatedScheduledExecutorService")) {
                 monitor(registry, unwrapThreadPoolExecutor(executorService, executorService.getClass()));
             }
+            // For Executors.newSingleThreadExecutor()
             else if (className.equals("java.util.concurrent.Executors$FinalizableDelegatedExecutorService")
                     || className.equals("java.util.concurrent.Executors$AutoShutdownDelegatedExecutorService")) {
                 monitor(registry,
                         unwrapThreadPoolExecutor(executorService, executorService.getClass().getSuperclass()));
             }
+            // For Executors.newThreadPerTaskExecutor()
             else if (className.equals(CLASS_NAME_THREAD_PER_TASK_EXECUTOR)) {
```

Comment:
Isn't this also for `Executors.newVirtualThreadPerTaskExecutor()`?

Reply:
@shakuzen Right, I added it in c942049.